zstackctl/zstackctl/ctl.py

```
@@ -10308,17 +10310,41 @@ def build_catalina_opts(self, heap):
         return '-Xmx%sG %s' % (heap, ' '.join(opts))
 
+    def update_global_config(self, category, name, value):
+        try:
+            db_hostname, db_port, db_user, db_password = ctl.get_live_mysql_portal()
+        except CtlError:
+            warn('cannot get database connection, skip updating GlobalConfig %s.%s' % (category, name))
+            return
+
+        if db_password:
+            cmd = ShellCmd('''mysql -u %s -p%s --host %s --port %s zstack -e "update GlobalConfigVO set value='%s' where name='%s' and category='%s'"'''
+                           % (db_user, shell_quote(db_password), db_hostname, db_port, value, name, category))
```

Ruff (0.15.6)

[error] Possible SQL injection vector through string-based query construction

(S608)